src/app/tests/TestCommandInteraction.cpp

```diff
@@ -107,30 +107,33 @@ namespace app {
 
 CommandHandler::Handle asyncCommandHandle;
 
-struct ForcedSizeBuffer
+class ForcedSizeBuffer : public app::DataModel::EncodableToTLV
```

Comment:
I'll either break the the test component of the PR out to a standalone follow up PR, or wait for #33772 to land first